app/portainer/home/EnvironmentList/KubeconfigButton.tsx

```diff
@@ -34,48 +81,154 @@ export function KubeconfigButton({ environments }: Props) {
       category: 'kubernetes',
     });
 
-    showKubeconfigModal(environments);
+    selection.clear();
+    setShowDialog(true);
+  }
+
+  function handleClose() {
+    setShowDialog(false);
+  }
+
+  function handleDownload() {
+    confirmKubeconfigSelection();
+  }
+
+  function isKubeconfigButtonVisible(environments: Environment[]) {
+    if (window.location.protocol !== 'https:') {
+      return false;
+    }
+    return environments.some((env) => isKubernetesEnvironment(env.Type));
+  }
+
+  function useKubeServiceExpiry() {
+    const { isSuccess, isError, data, error } = useQuery('', async () => {
```

Comment:
I suggest just return the `useQuery` result.

Reply:
Done